### PR TITLE
Move Replay resources + dom replay scripts into core; Disable Mitm

### PR DESCRIPTION
**Pull request by blakebyrnes**
This PR does some maintenance on Replay and Core.

1. Move replay scripts to Core API and serve them as resources. This allows Core to version scripts to the data, as well as to prepare us for being able to replay dom inside Core/scraping scripts
2. Allow the MITM to be disabled and capture resources accordingly
3. Compress the DOMChanges so they can be streamed more quickly
4. Add back AllContentLoaded internally - we had some tests that need to wait for load

### CODE REVIEW

File: core/injected-scripts/domReplayer.ts [hunk 1]
```diff
@@ -0,0 +1,410 @@
...
+    replayInteractions(...args: any[]);
+    getIsMainFrame?: () => boolean;
+    idMap: Map<number, Node>;
+    debugLogs: any[];
+    debugToConsole: boolean;
+    getNodeById(id: number): Node;
+  }
+}
+
+// copied since we can't import data types
```
Comment (calebjclark) [inline core/injected-scripts/domReplayer.ts:17]: Why can't data types be imported?

Comment (blakebyrnes) [reply core/injected-scripts/domReplayer.ts:17]: This is an injected script - gets run in the browser...

File: core/lib/FrameNavigationsObserver.ts [hunk 2]
```diff
@@ -162,14 +166,11 @@ export default class FrameNavigationsObserver {
     for (const state of this.navigations.top.stateChanges.keys()) {
       // don't resolve states for redirected
       if (state === LocationStatus.HttpRedirected) continue;
-      let recordedStatus = PipelineStatus[state as IPipelineStatus];
-
-      // use painting stable "order of pipeline" if the passed in state is load
+      let pipelineStatus = PipelineStatus[state as IPipelineStatus];
       if (state === LoadStatus.Load) {
-        recordedStatus = PipelineStatus.PaintingStable;
+        pipelineStatus = PipelineStatus.AllContentLoaded;
```
Comment (calebjclark) [inline core/lib/FrameNavigationsObserver.ts:171]: Why the switch from PaintingStable to AllContentLoaded?

File: core/injected-scripts/domReplayer.ts (hunk 1 above)
Comment (calebjclark) [reply core/injected-scripts/domReplayer.ts:17]: So it's converted to javascript, right? And typescript's enums are converted to a JS object.

File: core/lib/FrameNavigationsObserver.ts (hunk 2 above)
Comment (blakebyrnes) [reply core/lib/FrameNavigationsObserver.ts:171]: Painting Stable actually gets handled separately right above this snippet. This is assigning a pipeline status to the "load" event that gets stored in the load statuses.

**Pull request merged**